### PR TITLE
`private_dns_zone_virtual_network_link` waitLRO

### PR DESCRIPTION
**Pull request by shu-ying789**
linked to #15641

test result:
private1

### CODE REVIEW

File: internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go [hunk 1]
```diff
@@ -109,11 +109,6 @@ resource "azurerm_virtual_network" "test" {
   location            = azurerm_resource_group.test.location
   resource_group_name = azurerm_resource_group.test.name
   address_space       = ["10.0.0.0/16"]
-
-  subnet {
-    name           = "subnet1"
-    address_prefix = "10.0.1.0/24"
-  }
```
Comment (katbyte) [inline internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go:116]: why are we removing this from the tests?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go (hunk 1 above)
Comment (shu-ying789) [reply internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go:116]: @katbyte Deleted because of test failure.

Comment (katbyte) [reply internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go:116]: what test error was it fixing? what changed that is no longer required?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go (hunk 1 above)
Comment (shu-ying789) [reply internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go]: @katbyte The tests have been modified, please check again, thanks.

Comment (shu-ying789) [reply internal/services/privatedns/private_dns_zone_virtual_network_link_resource_test.go]: @katbyte I tried it without modifying tests and it works.

**Review by katbyte — APPROVED** [review]
Thanks @shu-ying789 - LGTM 🌻

**Pull request merged**